homeassistant/components/google/__init__.py

```diff
@@ -300,22 +300,27 @@ async def async_setup_services(
     calendars = await hass.async_add_executor_job(
         load_config, hass.config.path(YAML_DEVICES)
     )
+    calendars_file_lock = asyncio.Lock()
 
     async def _found_calendar(calendar_item: Calendar) -> None:
         calendar = get_calendar_info(
             hass,
             calendar_item.dict(exclude_unset=True),
         )
         calendar_id = calendar_item.id
-        # Populate the yaml file with all discovered calendars
-        if calendar_id not in calendars:
-            calendars[calendar_id] = calendar
-            await hass.async_add_executor_job(
-                update_config, hass.config.path(YAML_DEVICES), calendar
-            )
-        else:
-            # Prefer entity/name information from yaml, overriding api
-            calendar = calendars[calendar_id]
+        # If the google_calendars.yaml file already exists, populate it for
+        # backwards compatibility, but otherwise do not create it if it does
+        # not exist.
+        if calendars:
```

Comment:
Do the tests assert that if there already is a `google_calendars.yaml`, that is updated and if there isn't it is not?

Reply:
No, there were no explicit tests -- i was being lazy, but now i've added them.